src/tests/ibus-share.c

```diff
@@ -4,10 +4,19 @@
 static void
 test_machine_id (void)
 {
+    DBusError error = DBUS_ERROR_INIT;
     const gchar *s1 = ibus_get_local_machine_id ();
-    gchar *s2 = dbus_get_local_machine_id ();
+    gchar *s2 = dbus_try_get_local_machine_id (&error);
 
-    g_assert_cmpstr (s1, ==, s2);
+    g_assert (s1);
+    if (!s2) {
+        g_warning ("Failed to get locale machine id: %s: %s",
+                   error.name, error.message);
+        dbus_error_free (&error);
+        g_assert_not_reached ();
```

Comment:
If you're making the test crash out with a fatal error whenever `dbus_try_get_local_machine_id` fails, you could just do something like this:

```suggestion
        g_error ("Failed to get local machine id: %s: %s",
                   error.name, error.message);
```

Reply:
Thank you. I follow your suggestion.
I thought g_error() causes  the test crash without outputting the message in some cases but I cannot remember them.